### PR TITLE
[4.3] Vendor OpenShift controller manager observers

### PR DESCRIPTION
**Pull request by csrwng**
Makes it possible to configure build defaults, deployer/builder image, etc.

### CODE REVIEW

**Comment by csrwng** [comment]
Depends on #170

**Comment by csrwng** [comment]
@relyt0925 ptal

File: pkg/controllers/openshift_controller_manager/operator_client.go [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+	if err != nil {
+		return
+	}
+	var configBytes []byte
+	configBytes, err = mergeConfig([]byte(cm.Data["config.yaml"]), updateJSON)
+	if err != nil {
+		return
+	}
+	cm.Data["config.yaml"] = string(configBytes)
+	c.Logger.Info("Updating OpenShift APIServer configmap")
```
Comment (relyt0925) [inline pkg/controllers/openshift_controller_manager/operator_client.go]: wouldn't this one be controller manager configmap?

File: pkg/controllers/openshift_controller_manager/operator_client.go [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+	dataHash := calculateHash(configBytes)
+	var deployment *appsv1.Deployment
+	deployment, err = c.Client.AppsV1().Deployments(c.Namespace).Get(deploymentName, metav1.GetOptions{})
+	if err != nil {
+		return
+	}
+	if deployment.Spec.Template.ObjectMeta.Annotations == nil {
+		deployment.Spec.Template.ObjectMeta.Annotations = map[string]string{}
+	}
+	c.Logger.Info("Updating OpenShift APIServer deployment")
```
Comment (relyt0925) [inline pkg/controllers/openshift_controller_manager/operator_client.go]: same wouldn't this be controller manager?

**Comment by csrwng** [comment]
Thx! Fixed

**Comment by relyt0925** [comment]
holding just in case an accidental review comes in

**Pull request merged**